src/metabase/driver/sql/query_processor.clj

```diff
@@ -1715,16 +1715,22 @@
         desired-aliases (mapv second alias-info)
         duplicate-source-aliases? (and (> (count source-aliases) 1)
                                        (not (apply distinct? source-aliases)))
+        mismatched-aliases? (some #(and (string? (first %))
```

Comment:
probably worth adding a small test to `metabase.driver.sql.query-processor-test` just for this logic specifically too

Reply:
I added one to `metabase.query-processor-test.nested-queries-test`, but it's a bloodbath.

https://github.com/metabase/metabase/pull/47866/files#r1755557189

Reply:
This change has been reverted in favor of rolling back to the old `unique-name-generator` in the latest commit.